component/component.go

```diff
@@ -107,3 +107,25 @@ type ConfigUnmarshaler interface {
 // intoCfg interface{}
 //   An empty interface wrapping a pointer to the config struct to unmarshal into.
 type CustomUnmarshaler func(componentViperSection *viper.Viper, intoCfg interface{}) error
+
+// CreateParamsBase is passed to Create* functions.
+type CreateParamsBase struct {
```

Comment:
The problem is that we pass the Params by value and this will copy all the time. Is that expected, not worried about performance but about the behavior of having things copied.

Probably would be useful to have a NewExtensionCreateParams(component.WithApplicationStartInfo(), component.WithLogger()) to avoid user having to worry about the chaining created by having the base struct. Because we don't have this for the moment I would probably suggest to have this in every Param struct to allow easy initialization.